ambry-protocol/src/main/java/com.github.ambry.protocol/PutRequest.java

```diff
@@ -144,11 +144,13 @@ public long writeTo(WritableByteChannel channel) throws IOException {
       if (!bufferToSend.hasRemaining()) {
         written += channel.write(blob);
         if (!blob.hasRemaining()) {
-          blobWriteComplete = true;
+          if (currentVersion == Put_Request_Version_V3) {
+            okayToWriteCrc = true;
```

Comment:
minor: can replace with either
`okayToWriteCrc = !blob.hasRemaining() && currentVersion == Put_Request_Version_V3`
or just have both the conditions in on if